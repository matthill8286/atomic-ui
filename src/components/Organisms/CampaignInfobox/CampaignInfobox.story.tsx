import { radios, text } from '@storybook/addon-knobs'
import { storiesOf } from '@storybook/react'
import * as React from 'react'
import { StorybookRouterWrapper } from '@/utils/StorybookWrapper'
import { CampaignInfobox } from './CampaignInfobox'

storiesOf('Design System/Organisms/CampaignInfobox', module).add('CampaignInfobox', () => {
  const sampleText = text(
    'Example text',
    `Jetzt gratis liefern lassen ab einem Warenkorbwert von 100€. Für mehr Informationen hier klicken.`
  )
  const productImages: { [key: string]: string } = {
    none: '',
    Example1: 'public/images/teasers/product01.png',
    Example2: 'public/images/teasers/product02.png',
    Example3: 'public/images/teasers/versandkostenfrei.jpeg',
    Example4:
      'https://assets.mmsrg.com/is/166325/12975367df8e182e57044734f5165e190/c3/-/a2e1581609c914b0da60e8278c5aa062b?version=0&x=600&y=450&format=png&app=im&papp=quant(quality=55)',
  }
  const sampleImages = radios('Image', productImages, productImages['Example3'])

  return (
    <StorybookRouterWrapper>
      <CampaignInfobox text={sampleText} imgUrl={sampleImages} />
    </StorybookRouterWrapper>
  )
})
