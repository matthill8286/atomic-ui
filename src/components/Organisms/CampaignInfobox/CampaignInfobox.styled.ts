import { Link } from '@/components/Atoms/Link'
import { Picture } from '@/components/Atoms/Picture'
import { css, media, styled } from '@/styles'

export const StyledWrapper = styled.div(
  ({ theme }) => css`
    width: 100%;
    display: flex;
    background-color: ${theme.color.white};
    border-radius: ${theme.dimension.borderRadius1};
    border: 1px solid ${theme.color.grey2};
    &:hover {
      cursor: pointer;
      transform: translateY(-1px);
      box-shadow: 0 8px 10px 1px rgba(0, 0, 0, 0.07), 0 3px 14px 2px rgba(0, 0, 0, 0.06),
        0 5px 5px -3px rgba(0, 0, 0, 0.1);
      border-color: rgba(0, 0, 0, 0);
    }
  `
)

export const StyledLink = styled(Link)`
  width: 100%;
`

export const StyledTextWrapper = styled.div(
  () => css`
    display: flex;
    padding-left: 0.5rem;
    span {
      margin: auto;
      margin-right: 16px;
    }

    ${media.ie11} {
      align-items: center;
    }
  `
)

export const StyledPicture = styled(Picture)(
  () => css`
    display: flex;
    padding: 0.5rem;
    :hover {
      cursor: pointer;
    }
    img {
      max-height: 60px;
      width: auto;
    }
    picture {
      display: flex;
    }

    ${media.ie11} {
      width: auto;
    }
  `
)
