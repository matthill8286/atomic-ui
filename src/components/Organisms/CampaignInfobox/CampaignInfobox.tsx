import * as React from 'react'
import { Heading } from '@/components/Atoms/Typography'
import { TeaserLink } from '../Teasers/Teasers.interface'
import {
  StyledLink,
  StyledPicture,
  StyledTextWrapper,
  StyledWrapper,
} from './CampaignInfobox.styled'

export interface CampaignInfoboxProps {
  text: string
  imgUrl?: string
  link?: TeaserLink
  onClick?: (e: React.MouseEvent) => void
}

export const CampaignInfobox: React.FC<CampaignInfoboxProps> = ({
  text,
  imgUrl,
  link,
  onClick,
}: CampaignInfoboxProps) => {
  return (
    <StyledLink to={link?.to} href={link?.href} onClick={onClick} underline={false}>
      <StyledWrapper>
        {imgUrl && <StyledPicture src={imgUrl} objectFit="contain" />}
        <StyledTextWrapper>
          <Heading tag="span" scale="level-5" weight="semibold" limitLines={2} color="grey5">
            {text}
          </Heading>
        </StyledTextWrapper>
      </StyledWrapper>
    </StyledLink>
  )
}
