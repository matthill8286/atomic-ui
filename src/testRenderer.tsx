import { mount, ReactWrapper } from 'enzyme'
import React from 'react'
import { MemoryRouter as Router } from 'react-router-dom'
import renderer from 'react-test-renderer'
import { ThemeProvider } from 'styled-components'
import { saiyanTheme } from '@/styles/sc-vars-saiyan'
import { Theme } from '@/types'

export function renderWithTheme<C extends React.Component, P = C['props'], S = C['state']>(
  tree: React.ReactElement<P>,
  theme = saiyanTheme
) {
  return renderer.create(<ThemeProvider theme={theme}>{tree}</ThemeProvider>).toJSON()
}

export function renderWithThemeAndRouter(tree: React.ReactElement, theme: Theme = saiyanTheme) {
  return renderer
    .create(
      <ThemeProvider theme={theme}>
        <Router>{tree}</Router>
      </ThemeProvider>
    )
    .toJSON()
}

export function renderWithRouter(tree: React.ReactElement) {
  return renderer.create(<Router>{tree}</Router>).toJSON()
}

export function mountWithTheme<C extends React.Component, P = C['props'], S = C['state']>(
  tree: React.ReactElement<P>
): ReactWrapper<P, S, C> {
  return mount(<ThemeProvider theme={saiyanTheme}>{tree}</ThemeProvider>)
}

export function mountWithThemeAndRouter<C extends React.Component, P = C['props'], S = C['state']>(
  tree: React.ReactElement<P>
): ReactWrapper<P, S, C> {
  return mount(
    <ThemeProvider theme={saiyanTheme}>
      <Router>{tree}</Router>
    </ThemeProvider>
  )
}
