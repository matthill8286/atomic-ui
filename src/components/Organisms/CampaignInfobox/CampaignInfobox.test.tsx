import React from 'react'
import { renderWithThemeAndRouter } from '@/testRenderer'
import { CampaignInfobox, CampaignInfoboxProps } from './CampaignInfobox'

describe('CampaignInfobox', () => {
  const mockCampaignInfoboxData: CampaignInfoboxProps = {
    text: 'Jetzt gratis liefern lassen ab einem Warenkorbwert von 100â‚¬.',
    imgUrl: 'public/images/teasers/versandkostenfrei.jpeg',
    link: { to: '#' },
  }
  it('renders correctly', () => {
    const tree = renderWithThemeAndRouter(<CampaignInfobox {...mockCampaignInfoboxData} />)
    expect(tree).toMatchSnapshot()
  })
})
