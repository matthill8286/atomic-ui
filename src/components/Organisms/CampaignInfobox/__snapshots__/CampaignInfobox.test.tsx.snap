// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CampaignInfobox renders correctly 1`] = `
.c6 {
  margin: 24px 0 8px 0;
  color: #666666;
  font-size: 18px;
  line-height: 20px;
  font-family: OswaldLight,Arial,sans-serif;
  font-weight: 600;
  -webkit-letter-spacing: 0.2px;
  -moz-letter-spacing: 0.2px;
  -ms-letter-spacing: 0.2px;
  letter-spacing: 0.2px;
  white-space: normal;
  overflow: hidden;
  max-height: calc(20px * 2);
  -webkit-line-clamp: 2;
  display: -webkit-box;
  -webkit-box-orient: vertical;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: -webkit-max-content;
  width: -moz-max-content;
  width: max-content;
  color: #999999;
  opacity: 1.0;
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: pointer;
  position: relative;
  padding: 16px;
}

.c0::after {
  content: '';
  position: absolute;
  display: none;
  bottom: -2px;
  height: 1px;
  background-color: #999999;
  -webkit-transform-origin: right center;
  -ms-transform-origin: right center;
  transform-origin: right center;
  -webkit-transform: scale(0,1);
  -ms-transform: scale(0,1);
  transform: scale(0,1);
  -webkit-transition: -webkit-transform 0.25s cubic-bezier(0.37,0.31,0.2,0.85);
  -webkit-transition: transform 0.25s cubic-bezier(0.37,0.31,0.2,0.85);
  transition: transform 0.25s cubic-bezier(0.37,0.31,0.2,0.85);
}

.c0:hover::after {
  -webkit-transform-origin: left center;
  -ms-transform-origin: left center;
  transform-origin: left center;
  -webkit-transform: scale(1,1);
  -ms-transform: scale(1,1);
  transform: scale(1,1);
}

.c3 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  max-width: 100%;
}

.c3 img {
  object-fit: contain;
}

.c2 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  background-color: #FFFFFF;
  border-radius: 4px;
  border: 1px solid #E5E5E5;
}

.c2:hover {
  cursor: pointer;
  -webkit-transform: translateY(-1px);
  -ms-transform: translateY(-1px);
  transform: translateY(-1px);
  box-shadow: 0 8px 10px 1px rgba(0,0,0,0.07),0 3px 14px 2px rgba(0,0,0,0.06),0 5px 5px -3px rgba(0,0,0,0.1);
  border-color: rgba(0,0,0,0);
}

.c1 {
  width: 100%;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding-left: 0.5rem;
}

.c5 span {
  margin: auto;
  margin-right: 16px;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 0.5rem;
}

.c4:hover {
  cursor: pointer;
}

.c4 img {
  max-height: 60px;
  width: auto;
}

.c4 picture {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

@media screen and (-ms-high-contrast:active),screen and (-ms-high-contrast:none) {
  .c3 {
    max-width: 100%;
  }
}

@media screen and (-ms-high-contrast:active),screen and (-ms-high-contrast:none) {
  .c3 img {
    max-width: 100%;
    width: auto;
    height: auto;
  }
}

@media screen and (-ms-high-contrast:active),screen and (-ms-high-contrast:none) {
  .c5 {
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
  }
}

@media screen and (-ms-high-contrast:active),screen and (-ms-high-contrast:none) {
  .c4 {
    width: auto;
  }
}

<a
  className="c0 c1"
  href="/"
  onClick={[Function]}
>
  <div
    className="c2"
  >
    <div
      className="c3 c4"
      width="100%"
    >
      <picture
        style={
          Object {
            "height": undefined,
            "width": "100%",
          }
        }
      >
        <img
          alt="image"
          loading="eager"
          src="public/images/teasers/versandkostenfrei.jpeg"
          width="100%"
        />
      </picture>
    </div>
    <div
      className="c5"
    >
      <span
        className="c6"
        color="grey5"
        fontFamily="default"
        fontSize="md"
        spacing="half"
      >
        Jetzt gratis liefern lassen ab einem Warenkorbwert von 100â‚¬.
      </span>
    </div>
  </div>
</a>
`;
